### PR TITLE
feat: [scm-181]: adding rbac to code module

### PR DESCRIPTION
**Pull request by cjlee01**
### Summary

adding rbac to code module in ngui

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/20-rbac/interfaces/PermissionIdentifier.ts [hunk 1]
```diff
@@ -220,5 +220,12 @@ export enum PermissionIdentifier {
   DELETE_STREAMING_DESTINATION = 'core_streamingDestination_delete',
 
   //IDP Permisssions
-  IDP_SETTINGS_MANAGE = 'idp_idpsettings_manage'
+  IDP_SETTINGS_MANAGE = 'idp_idpsettings_manage',
+
+  //code permissions
```
Comment (tan-nhu) [inline src/modules/20-rbac/interfaces/PermissionIdentifier.ts:225]: `Harness Code` instead of `code`

File: src/modules/60-code/CodeApp.tsx [hunk 2]
```diff
@@ -16,6 +16,9 @@ import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
 import { global401HandlerUtils } from '@common/utils/global401HandlerUtils'
 import commonRoutes from '@common/RouteDefinitions'
 import routes from './RouteDefinitions'
+import { usePermission } from '@rbac/hooks/usePermission'
+
```
Comment (tan-nhu) [inline src/modules/60-code/CodeApp.tsx:20]: Remove blank line

File: src/modules/60-code/CodeApp.tsx [hunk 3]
```diff
@@ -70,6 +73,16 @@ const CODERemoteComponentMounter: React.FC<{
   )
   const history = useHistory()
   const { getToken: useGetToken } = SessionToken
+  const test = () => {
```
Comment (tan-nhu) [inline src/modules/60-code/CodeApp.tsx:76]: Why there's a `test` hook here?

File: src/modules/20-rbac/interfaces/PermissionIdentifier.ts (hunk 1 above)
Comment (cjlee01) [reply src/modules/20-rbac/interfaces/PermissionIdentifier.ts:225]: fixed

File: src/modules/60-code/CodeApp.tsx (hunk 2 above)
Comment (cjlee01) [reply src/modules/60-code/CodeApp.tsx]: fixed

File: src/modules/60-code/CodeApp.tsx (hunk 3 above)
Comment (cjlee01) [reply src/modules/60-code/CodeApp.tsx]: whoops i didn't push second commit that removed this @tan-nhu

**Comment by cjlee01** [comment]
retrigger titlecheck

**Review by tan-nhu — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

File: src/modules/60-code/CodeApp.tsx [hunk 4]
```diff
@@ -92,6 +93,20 @@ const CODERemoteComponentMounter: React.FC<{
   )
 }
 
+// return tooltip from here and can be more specific or generic
+function usePermissionTranslate(
+  usePermissionResult?: PermissionsRequest | undefined,
+  deps?: any[],
```
Comment (sahithikolichala) [inline src/modules/60-code/CodeApp.tsx:99]: can we make it string[] or something istead of any?

**Review by sahithikolichala — APPROVED** [review]

**Pull request merged**